### ISSUE TITLE
jfrog rt curl – "curl: option --url=https://example.com/: is unknown"

### ISSUE THREAD

**Issue by SimonAlling** [jfrog/artifactory-azure-devops-extension#186]
**Describe the bug**

The `JfrogCli` task runs `curl` with (for example) `--url=https://example.com/`, which is not an option curl recognizes.

Note that the problem is the equality sign (`=`):

```console
$ curl --url https://example.com/
<!doctype html>
[...]
$ curl --url=https://example.com/
curl: option --url=https://example.com: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
```

**To Reproduce**

Task definition:

```yaml
      - task: JfrogCli@1
        displayName: Check repo
        inputs:
          artifactoryService: $(ARTIFACTORY_SERVICE_CONNECTION)
          command: jfrog rt curl -XGET /api/repositories/foobar
```

Pipeline output with system diagnostics enabled:

```
##[debug]command=jfrog rt curl -XGET /api/repositories/foobar
##[debug]********-****-****-****-************=https://example.com/
##[debug]********-****-****-****-************ auth param username = ***
##[debug]********-****-****-****-************ auth param password = ***
##[debug]********-****-****-****-************ auth param apitoken = undefined
##[debug]Executing cliCommand: /opt/hostedtoolcache/jfrog/1.38.4/x64/jfrog rt curl -XGET /api/repositories/foobar --url="https://example.com/" --user=*** --password=***
curl: option --url=https://example.com/: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
[Error] exit status 2
##[debug]task result: Failed
##[error]Error: Command failed: /opt/hostedtoolcache/jfrog/1.38.4/x64/jfrog rt curl -XGET /api/repositories/foobar --url="https://example.com/" --user=*** --password=***
```

OK, maybe if we supply the `--url` flag (although it wouldn't be a proper solution):

```diff
       - task: JfrogCli@1
         displayName: Check repo
         inputs:
           artifactoryService: $(ARTIFACTORY_SERVICE_CONNECTION)
-          command: jfrog rt curl -XGET /api/repositories/foobar
+          command: jfrog rt curl -XGET /api/repositories/foobar --url https://example.com
```

No:

```diff
-##[debug]command=jfrog rt curl -XGET /api/repositories/foobar
+##[debug]command=jfrog rt curl -XGET /api/repositories/foobar --url https://example.com
 ##[debug]********-****-****-****-************=https://example.com/
 ##[debug]********-****-****-****-************ auth param username = ***
 ##[debug]********-****-****-****-************ auth param password = ***
 ##[debug]********-****-****-****-************ auth param apitoken = undefined
-##[debug]Executing cliCommand: /opt/hostedtoolcache/jfrog/1.38.4/x64/jfrog rt curl -XGET /api/repositories/foobar --url="https://example.com/" --user=*** --password=***
+##[debug]Executing cliCommand: /opt/hostedtoolcache/jfrog/1.38.4/x64/jfrog rt curl -XGET /api/repositories/foobar --url https://example.com --url="
 curl: option --url=https://example.com/: is unknown
 curl: try 'curl --help' or 'curl --manual' for more information
 [Error] exit status 2
 ##[debug]task result: Failed
-##[error]Error: Command failed: /opt/hostedtoolcache/jfrog/1.38.4/x64/jfrog rt curl -XGET /api/repositories/foobar --url="https://example.com/" --user=*** --password=***
+##[error]Error: Command failed: /opt/hostedtoolcache/jfrog/1.38.4/x64/jfrog rt curl -XGET /api/repositories/foobar --url https://example.com --url="
```

<del>So apparently, **JFrog CLI** adds the equality sign, because the final generated `jfrog rt curl` command above doesn't contain it. Also note the weird `--url="` at the end of the generated command.</del>

After some more digging, I'm not so sure anymore that the problem is in JFrog CLI. [This line in the **task code**](https://github.com/jfrog/artifactory-azure-devops-extension/blob/c869626cf731e5bb10f0b373527a4e3384a01277/artifactory-tasks-utils/utils.js#L315) seems like it could be the culprit:

```javascript
    cliCommand = cliJoin(cliCommand, '--url=' + quote(artifactoryUrl));
```

**Expected behavior**

The task should succeed.

**Versions**
- Artifactory Azure DevOps extension version: 1.10.2
- Agent operating system: Ubuntu 18.04.5
- Artifactory Version: N/A
- curl version: 7.58.0

**Comment by vyckou** [comment]
I think problem is a bit different here. This extension adds  --url, --user and --password flags/arguments to any given `jfrog rt ` task, while only part of the commands supports those 3 flags.
this issue is because `jfrog rt curl` does not accept `--url` , only `--server-id`
```
➜  source git:(master) ✗ jfrog rt curl                                               
[Error] Wrong number of arguments. You can read the documentation at https://www.jfrog.com/confluence/display/CLI/JFrog+CLI

Name:
  jfrog rt curl - Execute a cUrl command, using the configured Artifactory details.

Usage:
  jfrog rt curl [command options] <curl command>

Arguments:
  curl command
    cUrl command to run.

Options:
  --server-id    [Optional] Artifactory server ID configured using the config command.
```
where for example, jfrog command with url, username and password flag support
```
➜  source git:(master) ✗ jfrog rt build-publish --help

Name:
  jfrog rt build-publish - Publish build info.

Usage:
  jfrog rt bp [command options] <build name> <build number>

Arguments:
  build name
    Build name.

  build number
    Build number.

Options:
  --access-token      [Optional] Artifactory access token.
  --apikey            [Optional] Artifactory API key.
  --build-url         [Optional] Can be used for setting the CI server build URL in the build-info.
  --dry-run           [Default: false] Set to true to get a preview of the recorded build info, without publishing it to Artifactory.
  --env-exclude       [Default: *password*;*psw*;*secret*;*key*;*token*] List of case insensitive patterns in the form of "value1;value2;...". Environment variables match those patterns will be excluded.
  --env-include       [Default: *] List of patterns in the form of "value1;value2;..." Only environment variables match those patterns will be included.
  --insecure-tls      [Default: false] Set to true to skip TLS certificates verification.
  --password          [Optional] Artifactory password.
  --server-id         [Optional] Artifactory server ID configured using the config command.
  --ssh-key-path      [Optional] SSH key file path.
  --ssh-passphrase    [Optional] SSH key passphrase.
  --url               [Optional] Artifactory URL.
  --user              [Optional] Artifactory username.
```
in addition, for consistency, that those url, user and password flags should be added right after `jfrog rt command` and not at the end of the command

**Comment by SimonAlling** [comment]
Just want to connect this issue to #191, whose discussion contains relevant information.

**Comment by talarian1** [comment]
Hi @SimonAlling,
Thanks for reporting this issue. Please be noted that our new [1.11.0 release](https://github.com/jfrog/artifactory-azure-devops-extension/releases/tag/1.11.0) includes a general fix(#213) for the discussed issue. The JFrog CLI task now configures a new server, makes it the default one and deletes it at the end. So no server ID is needed to be supplied and all commands should be executed properly.
We would appreciate any feedback from you.

**Comment by RobiNino** [comment]
Hi @SimonAlling , @vyckou , 
Im closing this issue as it should be resolved in release 1.11.0.
Do let us know if you have any new feedback about the matter.
Thanks!
